source/io/module_javaScriptInvoke.js

```diff
@@ -484,24 +440,19 @@
             errorTypeRef,
             errorDataRef) {
             var errorValueRef = Module.eggShell.createValueRef(errorTypeRef, errorDataRef);
+            var functionName = Module.eggShell.dataReadString(functionNamePointer);
+            var parameters = new Array(0);
 
-            var newError = {
-                status: false,
-                code: ERRORS.NO_ERROR.CODE,
-                source: ERRORS.NO_ERROR.MESSAGE
-            };
+            var returnValueRef = createValueRefFromPointerArray(returnPointer, 0);
+            var shouldReturnValueBeUpdated = shouldUpdateReturnValue(isInternalFunction, returnValueRef.typeRef);
+            if (!shouldReturnValueBeUpdated) {
+                parameters = parameters.concat(returnValueRef);
```

Comment:
This will vary the JS function signature (number of parameters) depending on if the return value is wired in which is a bit strange.

Instead, if the function is internal we should always prepend the returnValueRef. And if the returnValueRef is unwired then we should prepend `undefined`. That would be direct createValueRef returned undefined for this case as it did previously.

Reply:
Done.